### PR TITLE
Improvements for dex auth

### PR DESCRIPTION
**Pull request by maxmarkus**
Improvements which are required for console. 

- allow window.location.origin to be the redirect_uri (prevent routing if access_token is in the url)
- resilience for invalid iframe (on logout, re-login and)
- context switcher is now hidden if autologin is disabled and user logged out
- refactored backdrop due to some oncreate side effects, seems like there is a bug in svelte when using string concatenation inside template

### CODE REVIEW

File: core/src/services/routing.js [hunk 1]
```diff
@@ -288,6 +288,12 @@ const buildRoute = (node, path, params) =>
     : buildRoute(node.parent, `/${node.parent.pathSegment}${path}`, params);
 
 export const handleRouteChange = async (path, component, node, config) => {
+  if (
+    window.location.href.indexOf('access_token') !== -1 ||
```
Comment (pekura) [inline core/src/services/routing.js]: 1) Can we do this check a bit more specific? At least indexOf('access_token=').
2) Can we make it configurable? Like list of regex expressions for paths to be ignored by Luigi routing? That way the Luigi app owner would have full control over it.

Comment (maxmarkus) [reply core/src/services/routing.js]: Yes, thats a good idea :-)

File: docs/navigation-configuration.md [hunk 2]
```diff
@@ -212,6 +212,7 @@ You can configure the way Luigi tackles routing in your application in the **Rou
 
 - **useHashRouting** defines either hash-based (`url.com/#/yourpath`) or path-based (`url.com/yourpath`) routing.
 - **nodeParamPrefix** sets the prefix character when using the `LuigiClient.linkManager().withParam()` function, which provides a way to simply attach query parameters to the view URL for activities such as sorting and filtering. The URL contains the parameters to allow deep linking. If you want to use a different character prefix, define yours here. The default character is `~`.
+- **skipRoutingForUrlPatterns** defines regex patterns that should be skipped by the router when listening for path changes. Used for excluding redirect_uri parameters. Default patterns are `[/access_token=/, '/id_token=/]`
```
Comment (bszwarc) [inline docs/navigation-configuration.md]:
```suggestion
- **skipRoutingForUrlPatterns** defines regex patterns to be skipped by the router when listening for path changes. This parameter is used for excluding **redirect_uri** parameters. Default patterns are `[/access_token=/, '/id_token=/]`.
```

File: core/src/navigation/TopNav.html [hunk 3]
```diff
@@ -51,6 +53,7 @@
   import LogoTitle from './LogoTitle.html';
   import Authorization from '../Authorization.html';
   import { handleRouteClick } from '../services/routing.js';
+  import * as Navigation from './services/navigation.js';
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply core/src/navigation/TopNav.html]: done

File: core/src/Backdrop.html [hunk 4]
```diff
@@ -29,6 +25,24 @@
...
     },
+    onstate({ changed, current, previous }) {
+      if (changed.backdropActive) {
+        this.setBackdropClass();
+      }
+    },
+    methods: {
+      setBackdropClass() {
+        const baseClasses = 'fd-ui__overlay fd-overlay fd-overlay--modal ';
+        if (!this.get().backdropActive) {
```
[the remark this answers is not in the document]
Comment (maxmarkus) [reply core/src/Backdrop.html]: Less code, but more dependencies. Not a fan of this. I guess if we want to have it, we can write it by ourself with a small helper function. similar to `['class1', 'class2'].join(' ')`.

**Pull request merged**